code/parse/sexp.cpp

```diff
@@ -3758,6 +3771,164 @@ int get_sexp()
 	return start;
 }
 
+void register_new_container_index(const SCP_string &container_name, int container_index = -1)
+{
+	if (container_index == -1) {
+		container_index = (int)Sexp_containers.size() - 1;
+	}
+	Assert(container_index >= 0);
+	Assert(container_index < (int)Sexp_containers.size());
+	Container_indices_by_initial[SCP_tolower(container_name[0])].emplace_back(container_index);
+}
+
+bool stuff_one_generic_sexp_container(SCP_string& name, int& type, int& opf_type, SCP_vector<SCP_string>& data)
+{
+	bool valid = true; // try to skip past bad containers by not returning until end of function
+	SCP_string temp_type_string;
+
+	data.clear();
+
+	required_string("$Name:");
+	stuff_string(name, F_NAME);
+
+	if (name.empty()) {
+		Warning(LOCATION, "SEXP Container with empty name found");
+		log_printf(LOGFILE_EVENT_LOG, "SEXP Container with empty name found");
+		valid = false;
+	} else if (name.length() > sexp_container::NAME_MAX_LENGTH) {
+		Warning(LOCATION,
+			"SEXP Container name %s is longer than limit %u",
```

Comment:
Use `SIZE_T_ARG` instead of %u, here and elsewhere.  This is for cross-platform compatibility reasons.

Reply:
Thanks for pointing out this issue.

Since C++11, `"%zu"` is supported for `size_t`, meaning it's portable across any compiler we'd want to use, and IMO it's easier to read, so I'd rather use that.

See the big table at https://en.cppreference.com/w/cpp/io/c/fprintf

That said, if SCP guidelines require using `SIZE_T_ARG` here, I'll switch to that.

Reply:
@asarium would be able to explain in more detail, but yes, this has been SCP policy for a while now.  And in the FSO codebase, `SIZE_T_ARG` is replaced by `"%zu"` on most platforms.

Reply:
Actually, I'd forgotten that `sexp_container::NAME_MAX_LENGTH` is of type `int` and not `size_t`, so I'm now using `%d` to format it.